File: erlang/grade-school/grade_school.erl
-module(grade_school).
-export([new/0, add/3, get/2, sort/1]).

new() -> orddict:new().

get(Grade, School) ->
  case orddict:find(Grade, School) of
    error          -> [];
    {ok, Students} -> Students
  end.

add(Name, Grade, School) ->
  Students = get(Grade, School),
  orddict:store(Grade, [Name| Students], School).

sort(School) -> School.

File: ruby/grade-school/grade_school.rb
class School
  VERSION = 1
  def initialize
    @school = {}
  end

  def add(name, grade)
    (@school[grade] ||= []) << name
  end

  def grade(grade)
    (@school[grade] || []).sort
  end

  def to_h
    Hash[@school.sort].each { |k,v| v.sort! }
  end
end
